src/bcf/report/report.rs

```diff
@@ -0,0 +1,527 @@
+    let mut gene_consequence_data = HashMap::new();
+    let mut clin_sig_data = Vec::new();
+    let mut gene_clin_sig_data = HashMap::new();
+    let mut af_data = Vec::new();
+    let mut gene_af_data = HashMap::new();
+    let mut unique_genes = HashMap::new();
+    for (sample, path) in sample_calls.iter().sorted() {
+        let mut genes = HashMap::new();
+        let mut impacts = HashMap::new();
+        let mut gene_impacts = HashMap::new();
+        let mut consequences = HashMap::new();
+        let mut gene_consequences = HashMap::new();
+        let mut clin_sigs = HashMap::new();
+        let mut gene_clin_sigs = HashMap::new();
+        let mut ann_indices = HashMap::new();
+        let mut bcf_reader = bcf::Reader::from_path(path)?;
+        let header = bcf_reader.header().clone();
+        let mut sample_names = Vec::new();
+        for s in header.samples() {
+            sample_names.push(String::from_utf8(s.to_owned()).unwrap());
+        }
+        let header_records = header.header_records();
+        let ann_fields: Vec<_> = get_ann_description(header_records).unwrap();
+
+        for (i, field) in ann_fields.iter().enumerate() {
+            ann_indices.insert(field, i);
+        }
+
+        for res in bcf_reader.records() {
+            let mut record = res?;
+            let alleles = record
+                .alleles()
+                .into_iter()
+                .map(|allele| allele.to_owned())
+                .collect_vec();
+            let alt_alleles = &alleles[1..];
+            let ref_allele = alleles[0].to_owned();
+
+            let allel_frequencies = record.format(b"AF").float()?[0].to_vec();
+
+            let ann = record.info(b"ANN").string()?;
+            if let Some(ann) = ann {
+                for alt_allele in alt_alleles {
+                    let variant = if alt_allele == b"<DEL>" {
+                        "DEL"
+                    } else if alt_allele == b"<BND>" {
+                        "BND"
+                    } else if alt_allele == b"<INV>" {
+                        "INV"
+                    } else if alt_allele == b"<DUP>" {
+                        "DUP"
+                    } else if alt_allele.len() == 1 && ref_allele.len() == 1 {
+                        "SNV"
+                    } else if alt_allele.len() == ref_allele.len() {
+                        "MNV"
+                    } else {
+                        "Complex"
+                    };
+
+                    for entry in &ann {
+                        let fields: Vec<_> = entry.split(|c| *c == b'|').collect();
+                        let alt = &fields[0];
+                        if alt != &alt_allele.as_slice() {
+                            continue;
+                        }
+
+                        let impact = str::from_utf8(
+                            fields[*ann_indices.get(&String::from("IMPACT")).expect("No field named IMPACT found. Please only use VEP-annotated VCF-files.")],
```

Comment:
Shouldn't this work as well?
```suggestion
                            fields[*ann_indices.get("IMPACT").expect("No field named IMPACT found. Please only use VEP-annotated VCF-files.")],
```

Reply:
To avoid redundancy here, put this snippet into a closure, and just call it with different field names.

Reply:
I tried using only `IMPACT` but I am getting a type error. I added the closure and makes the code a lot shorter! Thanks for the suggestion!